### ISSUE TITLE
Image controller not recognizing latest image when using pre-release

### ISSUE THREAD

**Issue by maberny** [fluxcd/image-reflector-controller#120]
We have docker tags like `v{semver}-{commit-offset}-{gitHash}` generated by `git describe`. The issue is once stable tags of the format `v{semver}` get pushed to the repository,  tags that should be more recent do not get recognized.

```
v0.0.0-1-abcd (first tag)
v0.0.1 (second recognized tag)
v0.0.1-1-abcd (not recognized even if latest)
```

Here is an example of the imagePolicy spec: 
```
  policy:
    semver:
      range: '>=0.0.0-0'
  filterTags:
    pattern: '^v.*'
```

Let me know if i'm doing something wrong or need to provide more details.

I've also tested the case with the `"github.com/Masterminds/semver/v3"` package here is a link to the go playground: https://play.golang.org/p/M3UfEk1O8tZ

**Comment by maberny** [comment]
Something interesting to note is that having ` pattern: '^v(.*)-(.*)'` recognizes prerelease tags as latest

**Comment by stefanprodan** [comment]
Check out the semver spec: https://semver.org/

TLDR: v0.0.1 > v0.0.1-1-abcd you need to bump the patch version e.g. v0.0.2-1

**Comment by maberny** [comment]
You are right this is what we ended up doing. Thank you!
